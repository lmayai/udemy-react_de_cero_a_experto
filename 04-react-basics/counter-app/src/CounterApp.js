import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value = 10}) => {

  const [counter, setCounter] = useState(value);

  // handleAdd: usar la palabra handle es muy apropiado
  const handleAdd = (e) => {
    setCounter(counter+1);
    // setCounter( (c) => c + 1 )
  }

  const handleReset = (e) => setCounter(value);
  
  const handleSubtract = (e) => setCounter(counter-1);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={ handleAdd }>+</button>
      <button onClick={ handleReset }>Reset</button>
      <button onClick={ handleSubtract }>-</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number
}

export default CounterApp;