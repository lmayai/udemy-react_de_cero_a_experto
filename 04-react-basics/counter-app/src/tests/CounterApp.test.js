import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Testing on <CounterApp />', () => {
  
  let wrapper = shallow(<CounterApp/>);

  beforeEach(() => {
    wrapper = shallow(<CounterApp/>);
  })

  test('should create the snapshot of <Counter App />', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  test('should show default value of 100', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value}/>)
    const valueRendered = Number(wrapper.find('h2').text().trim());
    expect(value).toBe(valueRendered);
  })
  
  test('Should increment value when click plus button', () => {
    wrapper.find('button').at(0).simulate('click'); 
    const newValue = wrapper.find('h2').text().trim();
    expect(newValue).toBe('11');
  });

  test('should decrement value when click minus button', () => {
    wrapper.find('button').at(2).simulate('click');
    const newValue = wrapper.find('h2').text().trim();
    expect(newValue).toBe('9');
  })

  test('should reset value', () => {
    const value = 105;
    const wrapper = shallow(<CounterApp value={value}/>)
    wrapper.find('button').at(0).simulate('click'); // Suma 1
    wrapper.find('button').at(0).simulate('click'); // Suma 1
    wrapper.find('button').at(1).simulate('click'); // Reset
    const newValue = wrapper.find('h2').text().trim();
    expect(newValue).toBe('105');
  })
  
  
})
