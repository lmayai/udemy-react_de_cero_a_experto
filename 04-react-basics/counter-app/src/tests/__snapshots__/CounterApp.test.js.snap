// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <CounterApp /> should create the snapshot of <Counter App /> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</Fragment>
`;
